### PR TITLE
Update CMake build

### PR DESCRIPTION
**Pull request by software-made-easy**
Updated CMake build (added Qt6 support). Currently only works for static builds.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,7 +2,7 @@
 *.log
 *.dSYM
 *.plist
-*.pro.user
+*.user
```
Comment (software-made-easy) [inline .gitignore:5]: Also ignore ".txt.tuser"

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,47 +1,66 @@
 cmake_minimum_required(VERSION 3.3)
-project(qmarkdowntextedit)
+project(qmarkdowntextedit LANGUAGES CXX)
 
 #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+#set(CMAKE_CXX_STANDARD 11) Use this instead
+#set(CMAKE_CXX_STANDARD_REQUIRED ON)
```
Comment (software-made-easy) [inline CMakeLists.txt:7]: To require a C++ standard, you must uncomment the 2 lines below

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,47 +1,66 @@
 cmake_minimum_required(VERSION 3.3)
-project(qmarkdowntextedit)
+project(qmarkdowntextedit LANGUAGES CXX)
```
Comment (software-made-easy) [inline CMakeLists.txt:2]: Require only c++ compiler

**Comment by pbek** [comment]
@Waqar144, any comments?

File: qmarkdowntextedit-headers.pri [hunk 4]
```diff
@@ -6,4 +6,3 @@ HEADERS += \
     $$PWD/qmarkdowntextedit.h \
     $$PWD/qownlanguagedata.h \
     $$PWD/qplaintexteditsearchwidget.h
-    $$PWD/qownlanguagedata.h
```
Comment (software-made-easy) [inline qmarkdowntextedit-headers.pri:9]: File added twice, see line 7

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,47 +1,66 @@
...
     qplaintexteditsearchwidget.cpp
-    )
+    media.qrc
+)
 
-add_executable(qmarkdowntextedit ${SOURCE_FILES} ${RESOURCE_ADDED})
+add_library(qmarkdowntextedit ${QMARKDOWNTEXTEDIT_FILES})
 
-include_directories(${Qt5Widgets_INCLUDES})
+target_link_libraries(qmarkdowntextedit Qt${QT_VERSION_MAJOR}::Widgets Qt${QT_VERSION_MAJOR}::Svg ${INTL_LDFLAGS})
```
Comment (Waqar144) [inline CMakeLists.txt:53]: Why link with Qt svg?

**Comment by Waqar144** [comment]
Besides the extra svg dependency, looks okay.

File: CMakeLists.txt [hunk 6]
```diff
@@ -1,47 +1,66 @@
...
+# find qt
+find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets Svg)
+find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets Svg)
 
-set(RESOURCE_FILES
-        media.qrc
-        )
+# needed for windows
+if(WIN32)
+    set(INTL_LDFLAGS -lintl)
```
Comment (Waqar144) [inline CMakeLists.txt:22]: Are you sure this is needed here?

File: CMakeLists.txt (hunk 5 above)
Comment (software-made-easy) [reply CMakeLists.txt:53]: Because QPlainTextEditSearchWidget loads Svg images

File: CMakeLists.txt (hunk 6 above)
Comment (software-made-easy) [reply CMakeLists.txt:22]: I don't have Windows and never tested it, but when you create a new CMake Project in Qt Creator, it's there. So no i'm not.

File: CMakeLists.txt (hunk 5 above)
Comment (Waqar144) [reply CMakeLists.txt:53]: Does loading svg image means we have to explicitly link to qt svg? I would think not. Qt svg would be required if we used qt svg apis

Comment (software-made-easy) [reply CMakeLists.txt]: I tested it and I was wrong. I have now removed it.

**Review by Waqar144 — APPROVED** [review]

**Comment by pbek** [comment]
Great, thank you, @software-made-easy and @Waqar144!

**Pull request merged**